watertap/property_models/multicomp_aq_sol_prop_pack.py

```diff
@@ -1186,8 +1246,81 @@ def rule_visc_k_phase(b, p):
     def _radius_stokes_comp(self):
         add_object_reference(self, "radius_stokes_comp", self.params.radius_stokes_comp)
 
+    def _molar_volume_comp(self):
+        add_object_reference(self, "molar_volume_comp", self.params.molar_volume_comp)
+
     def _diffus_phase_comp(self):
-        add_object_reference(self, "diffus_phase_comp", self.params.diffus_phase_comp)
+        if self.params.config.diffus_calculation == DiffusivityCalculation.none:
+            add_object_reference(
+                self, "diffus_phase_comp", self.params.diffus_phase_comp
```

Comment:
In the future, we might want to consider making diffusivity (and the other parameters that are meant to be populated by user data) a variable that is immediately fixed after declaration, setting things up for parameter estimation later on if desired. 

That is beyond the scope of this PR.

Reply:
I am in favor of this and should be able to make it happen quickly in the next push. we are making it be `var` in the calculation method already.

Reply:
Agreed. Since it'll likely require changes across more files (tests), I figured this incremental change would be a good stepping stone for the time being.